prime-router/metadata/fhir_mapping/hl7/segments/NK1/RelatedPerson.yml

```diff
@@ -0,0 +1,314 @@
+resourceType: RelatedPerson
+
+# Not implementing NK1.10: not in mapping, not in NIST
+# Not implementing NK1.11: not in mapping, not in NIST
+# Not implementing NK1.14: not in mapping, not in NIST
+# Not implementing NK1.17: not in mapping, not in NIST
+# Not implementing NK1.18: not in mapping, not in NIST
+# Not implementing NK1.19: not in mapping, not in NIST
+# Not implementing NK1.21: not in mapping, not in NIST
+# Not implementing NK1.22: not in mapping, not in NIST
+# Not implementing NK1.23: not in mapping, not in NIST
+# Not implementing NK1.24: not in mapping, not in NIST
+# Not implementing NK1.25: not in mapping, not in NIST
+# Not implementing NK1.26: not in mapping, not in NIST
+# Not implementing NK1.27: not in mapping, not in NIST
+# Not implementing NK1.28: not in mapping, not in NIST
+# Not implementing NK1.29: not in mapping, not in NIST
+# Not implementing NK1.34: not in mapping, not in NIST
+# Not implementing NK1.35: not in mapping, not in NIST
+# Not implementing NK1.36: not in mapping, not in NIST
+# Not implementing NK1.38: not in mapping, not in NIST
+# Not implementing NK1.39: not in mapping, not in NIST
+
+id:
+  type: STRING
+  valueOf: "GeneralUtils.generateResourceId()"
+  expressionType: JEXL
+
+patient:
+  valueOf: datatype/Reference
+  expressionType: resource
+  specs: $Patient
+
+name:
+  expressionType: nested
+  generateList: true
+  expressions:
+    - valueOf: datatypes/XPN/HumanName
+      expressionType: resource
+      condition: $nk1-2 NOT_NULL
+      vars:
+        nk1-2: NK1.2
+      specs: NK1.2 *
+      constants:
+        hl7v2Name: "name"
```

Comment:
I don't think the repeats are working here. You need `generateList: true` on the expressions that are repeating too.